src/app/view/perspectives/chart/chart-perspective.component.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Answer Institute, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {AppState} from '../../../core/store/app.state';
+import {Store} from '@ngrx/store';
+import {selectDocumentsByQuery} from '../../../core/store/documents/documents.state';
+import {Subscription} from 'rxjs/Subscription';
+import {DocumentsAction} from '../../../core/store/documents/documents.action';
+import {selectNavigation} from '../../../core/store/navigation/navigation.state';
+import {QueryModel} from '../../../core/store/navigation/query.model';
+import {DocumentModel} from '../../../core/store/documents/document.model';
+import {Workspace} from '../../../core/store/navigation/workspace.model';
+import {Packer} from './model/collection-attribute-pair-packer';
+import {CollectionAttributePair} from './model/collection-attribute-pair';
+import {animate, style, transition, trigger} from '@angular/animations';
+
+@Component({
+  selector: 'chart-perspective',
+  templateUrl: './chart-perspective.component.html',
+  styleUrls: ['./chart-perspective.component.scss'],
+  animations: [
+    trigger('slide', [
+      transition(':enter', [
+        style({width: 0}),
+        animate(200, style({width: '*'}))
+      ]),
+      transition(':leave', [
+        animate(200, style({width: 0}))
+      ])
+    ])
+  ]
+})
+export class ChartPerspectiveComponent implements OnInit, OnDestroy {
+
+  private _documents: DocumentModel[];
+
+  get documents() {
+    return this._documents;
+  }
+
+  set documents(value) {
+    this._documents = value;
+    this.documentAttributes = this.usableAttributes();
+  }
```

Comment:
Complicated it for the sake of declarativness. Instead of adding the attribute recalculation next to every documents assignment (even though there is currently only one), just telling it to recalculate every time, the documents change.